### PR TITLE
Remove AspNet and AspNet.TelemetryHttpModule instrumentation projects

### LINKED ISSUE #3317: Move certain instrumentation projects to contrib repo
# Feature Request
- Move the following instrumentation projects to [contrib repo](https://github.com/open-telemetry/opentelemetry-dotnet-contrib):
   - OpenTelemetry.Instrumentation.AspNet
   - OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule
   - OpenTelemetry.Instrumentation.StackExchangeRedis

We would continue to have the following instrumentation projects in the main repo as they help us validate and run end-to-end tests for the SDK and these also serve as an example for other instrumentation projects to reference:

- OpenTelemetry.Instrumentation.AspNetCore
- OpenTelemetry.Instrumentation.GrpcNetClient
- OpenTelemetry.Instrumentation.Http
- OpenTelemetry.Instrumentation.SqlClient

### PR DESCRIPTION
**Pull request by utpilla**
Fixes #3317 

All the three packages that were planned to be moved out of the main repo have been moved to the contrib repo. I have also released a new package versioned `1.0.0-rc9.5` for all these three instrumentation projects from the contrib repo:

1. [OpenTelemetry.Instrumentation.AspNet](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNet)
2. [OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule)
3. [OpenTelemetry.Instrumentation.StackExchangeRedis](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.StackExchangeRedis)

## Changes
- Remove AspNet and AspNet.TelemetryHttpModule instrumentation projects and related content
- Update links in the README files to point to the contrib repo

### CODE REVIEW

**Review by reyang — APPROVED** [review]
Best PR of the day 😄

File: docs/metrics/customizing-the-sdk/README.md [hunk 1]
```diff
@@ -35,11 +35,10 @@ In a typical application, a single `MeterProvider` is created at application
 startup and disposed at application shutdown. It is important to ensure that the
 provider is not disposed too early. Actual mechanism depends on the application
 type. For example, in a typical ASP.NET application, `MeterProvider` is created
-in `Application_Start`, and disposed in `Application_End` (both methods part of
-Global.asax.cs file) as shown [here](../../../examples/AspNet/Global.asax.cs). In
-a typical ASP.NET Core application, `MeterProvider` lifetime is managed by
-leveraging the built-in Dependency Injection container as shown
-[here](../../../examples/AspNetCore/Program.cs).
+in `Application_Start`, and disposed in `Application_End` (both methods are a
+part of the Global.asax.cs file). In a typical ASP.NET Core application,
```
Comment (cijothomas) [inline docs/metrics/customizing-the-sdk/README.md:39]: can we link to the contrib repo example?

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 2]
```diff
@@ -64,7 +64,7 @@ the `ConfigureServices` of your `Startup` class. Refer to documentation for
 [OpenTelemetry.Instrumentation.AspNetCore](../OpenTelemetry.Instrumentation.AspNetCore/README.md).
 
 For an ASP.NET application, adding instrumentation is typically done in the
-`Global.asax.cs`. Refer to documentation for [OpenTelemetry.Instrumentation.AspNet](../OpenTelemetry.Instrumentation.AspNet/README.md).
+`Global.asax.cs`.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/README.md:67]: lets keep the link, but modify it to point to the new home.

**Review by cijothomas — APPROVED** [review]
LGTM.
Left suggestions to keep links to ASP.NET wherever possible, linking to new repo.

**Pull request merged**